### ISSUE TITLE
fix: `Client.forTestnet()` returns a `WebClient` instance with no mirror nodes set.

### ISSUE THREAD

**Issue by bguiz** [hashgraph/hedera-sdk-js#2263]
### Description

1. Edit the code below, listed under "Steps to reproduce" to replace `accountId` and `privateKey` with any actual account ID in Hedera Testnet, and corresponding private key.
2. Run the code in NodeJs
3. Separately, use the same code in a web application

Expected result, for both NodeJs and browser application:

- HCS topic subscription is successful, and HCS topic messages get returns as they get added to the HCS topic via the `callback`

Actual result, when run in NodeJs:

- ✅ Same as expected result

Actual result, when run in browser application:

- ❎ Error occurs because `client._mirrorNetwork.getNextMirrorNode()`
- ❓The object being initialised appears to **not** be a `Client` instance, but rather a `WebClient` instance
    - ![](https://i.stack.imgur.com/09Ys7.png)
    - perhaps there is some difference in how it should be initialised?

### Steps to reproduce

```js
import { Client, TopicMessageQuery } from '@hashgraph/sdk';

const accountId = '0.0.1521'; // actual account on Hedera Testnet
const privateKey = '302e02...'; // actual private key for 0.0.1521
const dummyClient = Client.forTestnet().setOperator(accountId, privateKey);

const subscription = new TopicMessageQuery()
    .setTopicId(topicId)
    .subscribe(dummyClient, callback);
```

## Refs

- Stackoverflow question, see "Attempt 2": https://stackoverflow.com/q/78328581/194982
- Discussion: https://swirldslabs.slack.com/archives/C0453UVBR42/p1713169189315929

### Hedera network

testnet

### Version

@hashgraph/sdk@2.42.0

### Operating system

macOS

**Comment by bguiz** [comment]
I've dug into the SDK code a bit further, and found that the `WebClient` implementation doesn't seem to initialise the mirror network, unlike the `NodeClient` implementation.

See `WebClient#setMirrorNetwork`:
https://github.com/hashgraph/hedera-sdk-js/blob/65a2323f/src/client/WebClient.js#L202-L210

And contrast that with `NodeClient#setMirrorNetwork`:
https://github.com/hashgraph/hedera-sdk-js/blob/65a2323f/src/client/NodeClient.js#L289C5-L312

Was it intentional to have these different implementations for `setMirrorNetwork`?
Or stated another way - is there something preventing the `BrowserClient` from accessing mirror nodes?

**Comment by pwoosam** [comment]
@bguiz I found this GitHub discussion related to the question. https://github.com/grpc/grpc-web/discussions/1348

tldr browsers won't support http2 frame fratures required for native gRPC, so servers need to run a grpc-web proxy. Only Hedera's consensus nodes currently run a gRPC-web proxy. So only the consensus nodes can support gRPC connections from the browser.

To support gRPC from the browser, mirror nodes will also need to deploy gRPC-web proxy.

Further explanation from Hedera in this project's readme: https://github.com/hashgraph/hedera-grpcWeb-proxy

**Comment by ivaylonikolov7** [comment]
Currently the web browsers don't support gRPC request. In order to support gRPC requests we need gRPC web proxies but the mirror node doesn't have such ones currently. We talked with the mirror node team and they said they have plans for that but there's no specific timeline for that so we have added alternatives for every other query. The only query that we don't have an web version alternative is the `TopicMessageQuery`. We could create an alternative that polls the Mirror Node every second but this would put a heavy load on the Mirror Node and we are not sure if this would be a good production-ready solution. We have added a [`react`  example ](https://github.com/hiero-ledger/hiero-sdk-js/tree/main/examples/frontend-examples/src/app/topic/message-query) that demonstrates how would a user use this `TopicMessageQuery` in a web environment if he has to. I think this is the best the SDK team can do currently. 

So I think we could close this issue now? @bguiz

**Comment by ivaylonikolov7** [comment]
2 weeks have passed to I am closing this issue due to inactivity. @bguiz you can reopen it if you don't like the current solution.
